client/components/QuestionsAndAnswers/QuestionsAndAnswers.jsx

```diff
@@ -25,16 +25,30 @@ const getQuestions = (id) => {
 const QuestionsAndAnswers = ( { productId }) => {
   const [questions, setQuestions] = useState([]);
   const [shownQuestions, setShownQuestions] = useState([]);
+  const [allQuestionsShown, setAllQuestionsShown] = useState(false);
 
   useEffect(() => {
     async function fetchQuestions() {
       const questionData = await getQuestions(productId);
       setQuestions(questionData.results);
-      setShownQuestions(questionData.results);
+      setShownQuestions(sortQuestionsByHelpfulness(questionData.results, 2));
     }
     fetchQuestions();
   }, []);
 
+  const handleMoreQuestionsClick = () => {
+    var numQuestionsToShow;
+    var allQuestionsShown;
+    if (questions.length > shownQuestions.length + 2) {
+      numQuestionsToShow = shownQuestions.length + 2;
+    } else {
+      numQuestionsToShow = questions.length;
+    }
+    var moreQuestions = sortQuestionsByHelpfulness(questions, numQuestionsToShow);
+    setShownQuestions(moreQuestions);
+    setAllQuestionsShown(true);
```

Comment:
What's the purpose of  setting ```setAllQuestionsShown``` to ```true``` here and not after line 45? The way I understand the current code, if a user clicks on ```more questions``` button, 2 more questions are displayed and ```allQuestionsShown``` is set to true. But what if there are more questions left to be displayed?